apps/api/src/server/applications/application/application.validators.js

```diff
@@ -0,0 +1,154 @@
+import { composeMiddleware } from '@pins/express';
+import { validateFutureDate, validatePastDate } from '@pins/platform';
+import { body, param } from 'express-validator';
+import { validationErrorHandler } from '../../middleware/error-handler.js';
+import * as caseRepository from '../../repositories/case.repository.js';
+import * as regionRepository from '../../repositories/region.repository.js';
+import * as serviceCustomerRepository from '../../repositories/service-customer.repository.js';
+import * as subSectorRepository from '../../repositories/sub-sector.repository.js';
+import * as zoomLevelRepository from '../../repositories/zoom-level.repository.js';
+
+/**
+ *
+ * @param {string} value
+ */
+const validateExistingSubsector = async (value) => {
+	const subSector = await subSectorRepository.getByName(value);
+
+	if (subSector === null) {
+		throw new Error('Unknown Sub-Sector');
+	}
+};
+
+/**
+ *
+ * @param {string[]} value
+ */
+const validateExistingRegions = async (value) => {
+	for (const regionName of value) {
+		const region = await regionRepository.getByName(regionName);
+
+		if (region === null) {
+			throw new Error('Unknown region');
+		}
+	}
+};
+
+/**
+ *
+ * @param {number} caseId
+ */
+const validateExistingApplication = async (caseId) => {
+	const application = await caseRepository.getById(caseId);
+
+	if (application === null) {
+		throw new Error('Unknown Application');
+	}
+};
+
+/**
+ *
+ * @param {number} value
+ * @param {{req: any}} requestInfo
+ */
+const validateExistingApplicantThatBelongsToCase = async (value, { req }) => {
+	const applicant = await serviceCustomerRepository.getById(value);
+
+	if (applicant === null) {
+		throw new Error('Unknown Applicant');
+	}
+
+	if (applicant.caseId !== Number.parseInt(req.params.id, 10)) {
+		throw new Error('Applicant does not belong to case');
+	}
+};
+
+/**
+ *
+ * @param {string} value
+ */
+const validateExistingMapZoomLevel = async (value) => {
+	const mapZoomLevel = await zoomLevelRepository.getByName(value);
+
+	if (mapZoomLevel == null) {
+		throw new Error('Unknown map zoom level');
+	}
+};
+
+/**
+ *
+ * @param {number} value
+ * @returns {Date}
+ */
+const timestampToDate = (value) => {
+	return new Date(value);
+};
+
+export const validateCreateUpdateApplication = composeMiddleware(
+	body('geographicalInformation.gridReference.easting')
+		.isLength({ min: 6, max: 6 })
+		.toInt()
+		.withMessage('Easting must be integer with 6 digits')
```

Comment:
are these messages meant to be shown to the user or they're just for our logs?

Reply:
These messages will be in the body of the response but not in out logs (as this is the expected behaviour.